app/models.py

```diff
@@ -1,11 +1,113 @@
-from app import db
+from app import myapp, login, db
 from datetime import datetime
+from time import time
 from sqlalchemy.types import Boolean, Date, DateTime, Float, Integer, Text, Time, Interval
 from werkzeug.security import generate_password_hash, check_password_hash
 from flask_login import UserMixin
+import jwt
+import base64
 
+# Set up the table User with different columns in Database:
 class User(db.Model, UserMixin):
+    # User's ID:
     id = db.Column(db.Integer, primary_key=True)
+    # Username:
+    username = db.Column(db.String(32), nullable=False)
+    # User's password:
+    password = db.Column(db.String(32), nullable=False)
+    # User's email:
+    email = db.Column(db.String(100), nullable=False, unique=True)
 
-    def __repr__(self): #for debugging process
-        return f'<user {self.id}: {self.username}, {self.fullname}>'
\ No newline at end of file
+    # Set up relationship with class Post:
+    posts = db.relationship('Post', backref='author', lazy='dynamic')
+
+    # Relationship with Reminders:
+    tasks = db.relationship("Task", backref="user", lazy="dynamic")
+
+    # Function which generates password hash for user's password --> more protected!
+    def set_password(self, password):
+        self.password = generate_password_hash(password)
+
+    # Function which checks the hash of password given by user and the password in Database:
+    def check_password(self, password):
+        return check_password_hash(self.password, password)
+
+    # Function which generates a reset password token for the user:
+    # The token will expire after 600 seconds (10 mins)
+    def get_reset_password_token(self, expires_in=600):
+        # Generates token by using the encode() method of PyJWT package: 
+        return jwt.encode(
+            # The dictionary below contains the data which will be encoded
+            # with our Flask app's secret key to sign the token
+            # and hashing algorithm 'HS256' for that token signature:
+            {'reset_password': self.id, 'exp': time() + expires_in},
+            myapp.config['SECRET_KEY'], algorithm='HS256')
+
+    # This static method verifies 1 reset password token then returns corresponding user:
+    @staticmethod
+    def verify_reset_password_token(token):
+        try:
+            # Decoding token by decode() method in PyJWT package:
+            id = jwt.decode(token, myapp.config['SECRET_KEY'], algorithms=['HS256'])['reset_password']
+        except:
+            # If the token is invalid (or has been already expired) --> None:
+            return
+        # If the token is valid --> the user object corresponding to the user id in the token.
+        return User.query.get(id)
+
+    # Function that returns 1 string every time creating 1 new row (user): 
+    def __repr__(self):
+        return f'<User {self.id}: {self.username}, {self.fullname}>'
+
+# Set up the table Post with different columns in Database:
+class Post(db.Model):
```

Comment:
What is this Post class for?

Reply:
I think it's the database to keep track the metadata of each user account like creating date or things like that, but it seems like we could delete it :)